nautobot/ipam/views.py

```diff
@@ -661,6 +677,98 @@ def post(self, request):
         )
 
 
+class IPAddressMergeView(view_mixins.GetReturnURLMixin, view_mixins.ObjectPermissionRequiredMixin, View):
+    queryset = IPAddress.objects.all()
+
+    def get_required_permission(self):
+        return get_permission_for_model(self.queryset.model, "change")
+
+    def get(self, request):
+        host_values = self.queryset.values("host").order_by().annotate(count=models.Count("host")).filter(count__gt=1)
+        if host_values:
+            item = host_values[0]
+            queryset = self.queryset.filter(host__in=[item["host"]])
+            return render(
+                request=request,
+                template_name="ipam/ipaddress_merge.html",
+                context={
+                    "queryset": queryset,
+                    "return_url": self.get_return_url(request),
+                },
+            )
+        else:
+            msg = "No duplicate IPs found."
+            messages.warning(request, msg)
+            return redirect(self.get_return_url(request))
+
+    def post(self, request):
+        logger = logging.getLogger(__name__)
+        collapsed_ips = IPAddress.objects.filter(pk__in=request.POST.getlist("pk"))
+        with transaction.atomic():
+            try:
+                _, deleted_info = collapsed_ips.delete()
```

Comment:
Personally I think it'd be preferable to avoid deleting the IP that currently exists in the selected target namespace and just update it, rather than deleting all and creating a new one entirely.